src/execution_algorithms/ltfb/regularized_evolution.cpp

```diff
@@ -0,0 +1,343 @@
+
+  // Get metric values
+  bool found_metric = false;
+  EvalType score = 0.f;
+  std::string metric_name;
+  for (const auto& met : m.get_metrics()) {
+    metric_name = met->name();
+    if (m_metrics.count(metric_name)) {
+      found_metric = true;
+      score += met->get_mean_value(execution_mode::tournament);
+      break;
+    }
+  }
+
+  // sanity check
+  if (!found_metric) {
+    LBANN_ERROR("could not find metric \"",
+                metric_name,
+                "\" "
+                "in model \"",
+                m.get_name(),
+                "\"");
+  }
+
+  m.make_data_store_preloaded(execution_mode::tournament);
+
+  // Clean up and return metric score
+  m.reset_mode(ctxt, original_mode);
+  dc.reset_mode(ctxt);
+  return score;
+}
+
+void RegularizedEvolution::select_next(model& m,
+                                       ltfb::ExecutionContext& ctxt,
+                                       data_coordinator& dc) const
+{
+  // Find the best model trainer
+  // Find the oldest model trainer (highest age)
+  // Copy the best model trainer to oldest model trainer, mutate it and set its
+  // age to 0
+
+  auto const& comm = *(m.get_comm());
+  const unsigned int num_trainers = comm.get_num_trainers();
+  const int trainer_id = comm.get_trainer_rank();
+  auto const step = ctxt.get_step();
+
+  std::vector<EvalType> sample_trainers(num_trainers);
+  if (comm.am_world_master()) {
+    for (unsigned int i = 0; i < num_trainers; i++)
+      sample_trainers[i] = i;
+    std::shuffle(sample_trainers.begin(),
+                 sample_trainers.end(),
+                 get_ltfb_generator());
+  }
+  comm.world_broadcast(comm.get_world_master(),
+                       sample_trainers.data(),
+                       num_trainers);
+
+  // If rank within first S, send score , else score = 0
+  auto it =
+    std::find(sample_trainers.begin(), sample_trainers.end(), trainer_id);
+
+  El::Int score = evaluate_model(m, ctxt, dc);
+  // If in sample, send true score
+  if (std::distance(sample_trainers.begin(), it) < m_sample_size) {
+    // score = evaluate_model(m, ctxt, dc);
+    if (comm.am_trainer_master())
+      std::cout << "Trainer " << trainer_id << " in sample" << std::endl;
+  }
+  // Else force score to 0
+  else {
+    // score = evaluate_model(m, ctxt, dc);
+    score = 0;
+  }
+
+  std::vector<EvalType> score_list(num_trainers);
+  comm.trainer_barrier();
+  if (comm.am_trainer_master()) {
+    comm.all_gather<EvalType>(score, score_list, comm.get_intertrainer_comm());
+  }
+  // Communicate trainer score list from trainer master to other procs in
+  // trainer
+  comm.trainer_broadcast(comm.get_trainer_master(),
+                         score_list.data(),
+                         num_trainers);
+
+  // Find winning trainer (in sample)
+  El::Int winner_id =
+    std::distance(score_list.begin(),
+                  std::max_element(score_list.begin(), score_list.end()));
```

Comment:
This will misbehave if lower metrics are better or if negative metric values are valid.

Reply:
Using a different logic now.